### PR TITLE
initialize in background thread

### PR DESCRIPTION
**Pull request by kenny-statsig**
Add ability to initialize in background thread. 
Also add an API to check if initialized in case you want to delay checks until then.

Ensure no race condition by killing any existing async thread.

### CODE REVIEW

File: lib/country_lookup.rb [hunk 1]
```diff
@@ -2,6 +2,9 @@ module CountryLookup
 
   class Lookup
     def initialize
+      if !@initialize_bg_thread.nil?
```
Comment (tore-statsig) [inline lib/country_lookup.rb]: this is a constructor in ruby, right? So wont the lookup ip string and lookup ip number functions still potentially create a new instance of this, kill in the background thread?

Comment (kenny-statsig) [reply lib/country_lookup.rb]: Oh I see what you're saying. This needs to live outside of the class.

File: lib/country_lookup.rb [hunk 2]
```diff
@@ -99,20 +99,35 @@ def initialize_from_file
...
+  end
+
+  def self.is_ready_for_lookup
+    !@lookup.nil?
+  end
+
   def self.lookup_ip_string(ip_string)
     if @lookup.nil?
-      @lookup = Lookup.new
+      initialize
```
Comment (tore-statsig) [inline lib/country_lookup.rb:124]: So this is still going to call initialize on the main thread if it hasnt finished and this gets used, right?  Still dont think this is right

Comment (kenny-statsig) [reply lib/country_lookup.rb:124]: We could wait on the original thread instead. Or were u proposing something else? Like return a custom eval reason until it's initialized

**Comment by tore-statsig** [comment]
Spoke offline.  As long as you make this threadsafe, dont kick off a new thread, we will do async init and block on the first check if initialization is not complete

**Review by tore-statsig — APPROVED** [review]

**Pull request merged**